src/audio/pipeline/pipeline-stream.c

```diff
@@ -431,8 +439,15 @@ int pipeline_trigger(struct pipeline *p, struct comp_dev *host, int cmd)
 
 			for (int i = 0; i < arch_num_cpus(); i++) {
 				if (data.kcps[i] > 0) {
+					uint32_t core_kcps = core_kcps_get(i);
+
+					/* Cannot go below 50000kcps */
+					if (data.kcps[i] > core_kcps - 50000)
+						data.kcps[i] = core_kcps - 50000;
```

Comment:
@lyakh  I think this has to be cleaned up. At least a magic number definition at start of the file and preferably a Kconfig as this is hardly a constant that covers all platforms.

I'm ok to do this as a follow-up, as this PR has so many moving parts.